app/lib/intervention.cjsx

```diff
@@ -0,0 +1,97 @@
+ExperimentsClient = require './experiments-client'
+config = require './intervention-config'
+{getSessionID} = require '../lib/session'
+
+Intervention = React.createClass
+
+  contextTypes:
+    geordi: React.PropTypes.object
+    interventionMonitor: React.PropTypes.object
+    experimentsClient: React.PropTypes.object
+
+  propTypes:
+    experimentName: React.PropTypes.string.isRequired
+    sessionID: React.PropTypes.string.isRequired
+    interventionID: React.PropTypes.string.isRequired
+    interventionDetails: React.PropTypes.object.isRequired
+
+  endIntervention: ->
+    @context.experimentsClient.postDataToExperimentServer @context.interventionMonitor,
+                                                 @context.geordi,
+                                                 @props.experimentName,
+                                                 @props.user.id,
+                                                 @props.sessionID,
+                                                 "intervention",
+                                                 @props.interventionID
+
+  skipIntervention: (event) ->
+    logData = {
+      sessID: @props.sessionID
+      ivnID: @props.interventionID
+      cancelled: true
+    }
+    @context.experimentsClient.logExperimentData @context.geordi, 'skipIntervention', logData
+    @endIntervention()
+
+  answerQuestion: (event) ->
+    logData = {
+      sessID: @props.sessionID
+      ivnID: @props.interventionID
+      answer: document.getElementsByClassName("intervention-question")[0].value
+    }
+    @context.experimentsClient.logExperimentData @context.geordi, 'interventionResponse', logData
+    @endIntervention()
+    event.target.disabled = true
+
+  endStatement: (event) ->
+    @endIntervention()
+    event.target.disabled = true
+
+  componentWillMount: ->
+    if @context.experimentsClient.currentExperimentState isnt @context.experimentsClient.EXPERIMENT_STATE_INTERVENTION_ON_SCREEN
+      @context.experimentsClient.logExperimentState @context.geordi, @context.interventionMonitor?.latestFromSugar, "interventionStart"
+
+  getBodyMarkup: (html) ->
+    { __html: html }
+
+  logLinkClick: (event) ->
+    logData = {
+      sessID: @props.sessionID
+      ivnID: @props.interventionID
+      linkAddress: event.target.getAttribute("href")
+    }
+    @context.experimentsClient.logExperimentData @context.geordi, 'interventionLinkClicked', logData
+
+  componentDidMount: ->
+    interventionBody = document.getElementsByClassName("intervention-body")[0]
+    linksInInterventions = interventionBody.getElementsByTagName("a")
+    for link in linksInInterventions
+      link.addEventListener("click", @logLinkClick)
+
+  render: ->
+    <div className="intervention">
+      <h3 className="intervention-title">{@props.interventionDetails.title}:</h3>
+      <p dangerouslySetInnerHTML={@getBodyMarkup(@props.interventionDetails.body)} className="intervention-body"/>
```

Comment:
Yes, some of the interventions have links, I haven't yet thought of a better way to insert them since it's variable. I understand Jim's points that `<Link>` would be preferable but not sure how to achieve this with variable data. I guess i could build a react function that produces the bodies wrapped in a `<p>` or something....